exonum-java-binding-common/src/test/java/com/exonum/binding/common/proofs/list/ListProofStructureValidatorTest.java

```diff
@@ -165,16 +165,32 @@ void visitUnbalancedElementNodeTooDeep() {
   }
 
   @Test
-  void visitUnbalancedHashNodeTooDeep() {
+  void visit_UnbalancedHashNodeTooDeep() {
     int depth = ListProofStructureValidator.MAX_NODE_DEPTH + 1;
-    ListProof root = generateLeftLeaningProofTree(depth, new ListProofHashNode(H2));
+    ListProof root = generateRightLeaningProofTree(depth, new ListProofHashNode(H2));
 
     validator = createListProofStructureValidator();
     root.accept(validator);
 
     assertThat(validator.check(), is(ListProofStatus.INVALID_HASH_NODE_DEPTH));
   }
 
+  @Test
+  void visit_UnbalancedHashNodesOnlyLeafs() {
+    ListProofBranch root = new ListProofBranch(
+        leafOf(V1),
```

Comment:
Please include hash here because at this moment this tree is not correct for _several_ reasons (unbalanced + a branch with two hashes)